specs/gloas/beacon-chain.md

```diff
@@ -750,113 +752,91 @@ def is_builder_payment_withdrawable(
     return builder.withdrawable_epoch >= current_epoch or not builder.slashed
 ```
 
-##### Modified `get_expected_withdrawals`
+##### New `get_builder_withdrawable_balance`
 
 ```python
-def get_expected_withdrawals(state: BeaconState) -> Tuple[Sequence[Withdrawal], uint64, uint64]:
-    epoch = get_current_epoch(state)
-    withdrawal_index = state.next_withdrawal_index
-    validator_index = state.next_withdrawal_validator_index
+def get_builder_withdrawable_balance(builder: Validator, balance: Gwei) -> Gwei:
+    """
+    Get the withdrawable balance for a builder payment.
+    """
+    if builder.slashed:
+        return balance
+    elif balance > MIN_ACTIVATION_BALANCE:
+        return balance - MIN_ACTIVATION_BALANCE
+    else:
+        return Gwei(0)
+```
+
+##### New `get_builder_withdrawals`
+
+```python
+def get_builder_withdrawals(
+    state: BeaconState,
+    withdrawal_index: WithdrawalIndex,
+    epoch: Epoch,
+    prior_withdrawals: Sequence[Withdrawal],
+) -> Tuple[Sequence[Withdrawal], WithdrawalIndex, uint64]:
     withdrawals: List[Withdrawal] = []
-    processed_partial_withdrawals_count = 0
-    processed_builder_withdrawals_count = 0
+    processed_count = 0
 
-    # [New in Gloas:EIP7732]
-    # Sweep for builder payments
     for withdrawal in state.builder_pending_withdrawals:
-        if (
-            withdrawal.withdrawable_epoch > epoch
-            or len(withdrawals) + 1 == MAX_WITHDRAWALS_PER_PAYLOAD
-        ):
+        all_withdrawals = prior_withdrawals + withdrawals
+        is_not_withdrawable = withdrawal.withdrawable_epoch > epoch
+        has_reached_bound = len(all_withdrawals) + 1 == MAX_WITHDRAWALS_PER_PAYLOAD
```

Comment:
`get_pending_partial_withdrawals`, `get_sweep_withdrawals` and `get_builder_withdrawals` have different max withdrawals bound. I think introducing a variable such as `withdrawals_limit` that represents the max withdrawal bound and comparing `has_reached_bound = len(all_withdrawals) == withdrawals_limit` would give us consistent approach, making it easier to read.